src/components/FileManager/InlineFilesSection.tsx

```diff
@@ -0,0 +1,350 @@
+/*!
+ * © 2021 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  getModelsByType,
+  hasObjectType,
+} from '@manuscripts/manuscript-transform'
+import {
+  ExternalFile,
+  Figure,
+  FigureElement,
+  Model,
+  ObjectTypes,
+  Section,
+  Table,
+  TableElement,
+} from '@manuscripts/manuscripts-json-schema'
+import React, { Dispatch, useCallback, useMemo } from 'react'
+import styled from 'styled-components'
+
+import { useDropdown } from '../../hooks/use-dropdown'
+import DotsIcon from '../icons/dots-icon'
+import {
+  FileDescription,
+  FileInfoContainer,
+  FileNameTitleContainer,
+  FileTitle,
+} from './FileSectionItem/FileInfo'
+import { ActionsIcon, Item } from './FileSectionItem/FileSectionItem'
+import { ItemActions } from './FileSectionItem/ItemActions'
+import { Action } from './FileSectionState'
+import {
+  extensionsWithFileTypesMap,
+  FileType,
+  fileTypesWithIconMap,
+} from './util'
+
+export interface ExternalFileRef {
+  url: string
+  kind?: string
+  ref?: ExternalFile
+}
+
+export const InlineFilesSection: React.FC<{
+  submissionId: string
+  modelMap: Map<string, Model>
+  handleReplace: (
+    submissionId: string,
+    name: string,
+    file: File,
+    typeId: string
+  ) => Promise<boolean>
+  handleDownload: (url: string) => void
+  dispatch: Dispatch<Action>
+}> = ({ submissionId, handleReplace, handleDownload, modelMap, dispatch }) => {
+  const inlineFiles = useMemo(() => {
+    const files: {
```

Comment:
can we move that function out into the lib to improve readability of that code, or maybe a custom hook?